mantidproject/mantid
# Project Recovery - Sanitise filename characters

Cannot be merged before #22781 

**Description of work.**
Removes any invalid characters that can exist in workspace names, such as `?` or `:`

**To test:**
- Create a sample workspace with any of the following characters: `? : \ / *`
- Check project recovery strips these characters out and saves .py file
- For example 123\abc -> 123abc

Fixes #22780 .

Does this update require release notes?

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
